From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Fri, 18 Oct 2024 17:01:06 +0200
Subject: fix: remove outdated V8 flags from node.cc

Refs https://chromium-review.googlesource.com/c/v8/v8/+/5507047
Refs https://chromium-review.googlesource.com/c/v8/v8/+/6249026

The above CL removes the `--harmony-import-assertions` and
--experimental-wasm-memory64 flags from V8.

This patch can be removed when we upgrade to a V8 version that
contains the above CLs.

diff --git a/src/node.cc b/src/node.cc
index 07684482f855363e26c3d7299a585a8a5654015e..627337efae49319e2a77b4686176ce92a8493024 100644
--- a/src/node.cc
+++ b/src/node.cc
@@ -816,7 +816,7 @@ static ExitCode ProcessGlobalArgsInternal(std::vector<std::string>* args,
   }
   // TODO(nicolo-ribaudo): remove this once V8 doesn't enable it by default
   // anymore.
-  v8_args.emplace_back("--no-harmony-import-assertions");
+  // v8_args.emplace_back("--no-harmony-import-assertions");
 
   auto env_opts = per_process::cli_options->per_isolate->per_env;
   if (std::find(v8_args.begin(), v8_args.end(),
@@ -828,7 +828,7 @@ static ExitCode ProcessGlobalArgsInternal(std::vector<std::string>* args,
 
   // Support stable Phase 5 WebAssembly proposals
   v8_args.emplace_back("--experimental-wasm-imported-strings");
-  v8_args.emplace_back("--experimental-wasm-memory64");
+  // v8_args.emplace_back("--experimental-wasm-memory64");
   v8_args.emplace_back("--experimental-wasm-exnref");
 
 #ifdef __POSIX__
